### PR TITLE
Feature/hero banner based on php7.4 branch

### PR DESCRIPTION
**Pull request by smoqadam**

### CODE REVIEW

File: src/ViewModel/HeroBanner.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace eLife\Patterns\ViewModel;
+
+use Assert\Assertion;
+use eLife\Patterns\ArrayAccessFromProperties;
+use eLife\Patterns\ArrayFromProperties;
+use eLife\Patterns\ViewModel;
+
+final class HeroBanner implements ViewModel
```
Comment (nlisgo) [inline src/ViewModel/HeroBanner.php:10]: We need a test for this. I would do a more thorough review but adding a test might catch some issues and simplify review. Look to the `CarouselItemTest` as an example of required coverage.

**Review by nlisgo — CHANGES_REQUESTED** [review]

File: src/ViewModel/HeroBanner.php (hunk 1 above)
Comment (smoqadam) [reply src/ViewModel/HeroBanner.php:10]: @nlisgo I just committed the test.

File: src/ViewModel/HeroBanner.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+{
+    use ArrayAccessFromProperties;
+    use ArrayFromProperties;
+
+    private $title;
```
Comment (nlisgo) [inline src/ViewModel/HeroBanner.php:15]: There is no way to set the `TitleLength` parameter which is part of the schema for the `hero-banner.mustache` template.

**Review by nlisgo — CHANGES_REQUESTED** [review]
I have made a few adjustments. We should introduce something `HasTitleLength` to determine which `titleLength` value to set.

See:
- https://github.com/elifesciences/patterns-php/blob/master/src/ViewModel/HasTitleLength.php
- https://github.com/elifesciences/patterns-php/blob/master/src/ViewModel/CarouselItem.php#L32

**Review by nlisgo — APPROVED** [review]

**Pull request merged**